packages/picasso/src/LexicalEditor/nodes/CustomEmojiNode.tsx

```diff
@@ -0,0 +1,141 @@
+import { TextNode } from 'lexical'
+import type {
+  DOMConversionMap,
+  DOMConversionOutput,
+  DOMExportOutput,
+  EditorConfig,
+  LexicalNode,
+  NodeKey,
+  SerializedTextNode,
+  Spread,
+} from 'lexical'
+
+export interface CustomEmojiPayload {
+  src: string
+  id: string
+}
+
+export type SerializedCustomEmojiNode = Spread<
+  {
+    src: string
+    id: string
+  },
+  SerializedTextNode
+>
+
+const convertImageElement = (domNode: Node): null | DOMConversionOutput => {
+  if (domNode instanceof HTMLImageElement) {
+    const src = domNode.getAttribute('src')
+    const id = domNode.getAttribute('data-emoji-name')
+
+    if (src && id) {
+      return {
+        node: $createCustomEmojiNode({
+          src,
+          id,
+        }),
+      }
+    }
+
+    return null
+  }
+
+  return null
+}
+
+export class CustomEmojiNode extends TextNode {
+  __src: string
+  __id: string
+
+  static getType() {
+    return 'custom-emoji'
+  }
+
+  static clone(node: CustomEmojiNode): CustomEmojiNode {
+    return new CustomEmojiNode(node.__src, node.__id)
+  }
+
+  constructor(src: string, id: string, key?: NodeKey) {
+    super(' ', key)
+    this.__src = src
+    this.__id = id
+  }
```

Comment:
There is a trick with typescript to auto-set fields from the constructor if you defined the visibility them so:
```ts
class CustomEmojiNode {
  src: string
  id: string
  
  constructor(src: string, id: string) {
    this.src = src
    this.id = id
  }
}
```

Can be written as:
```ts
class CustomEmojiNode {
  constructor(public src: string, public id: string) {
  }
}
```

Wdyt?
```suggestion
  constructor(public src: string, public id: string, key?: NodeKey) {
    super(' ', key)
  }
```